library/src/scala/quoted/Quotes.scala

```diff
@@ -3638,8 +3638,69 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
        *  @note As a macro can only splice code into the point at which it is expanded, all generated symbols must be
        *        direct or indirect children of the reflection context's owner.
        */
+      // TODO: add flags and privateWithin
       @experimental def newClass(parent: Symbol, name: String, parents: List[TypeRepr], decls: Symbol => List[Symbol], selfType: Option[TypeRepr]): Symbol
 
+      /** Generates a new module symbol with an associated module class symbol.
+       *  This returns the module symbol. The module class can be accessed calling `moduleClass` on this symbol.
+       *
+       *  Example usage:
+       *  ```scala
+       *  //{
+       *  given Quotes = ???
+       *  import quotes.reflect._
+       *  //}
+       *  val moduleName: String = Symbol.freshName("MyModule")
+       *  val parents = List(TypeTree.of[Object])
+       *  def decls(cls: Symbol): List[Symbol] =
+       *    List(Symbol.newMethod(cls, "run", MethodType(Nil)(_ => Nil, _ => TypeRepr.of[Unit]), Flags.EmptyFlags, Symbol.noSymbol))
+       *
+       *  val mod = Symbol.newModule(Symbol.spliceOwner, moduleName, Flags.EmptyFlags, Flags.EmptyFlags, parents.map(_.tpe), decls, Symbol.noSymbol)
+       *  val cls = mod.moduleClass
+       *  val runSym = cls.declaredMethod("run").head
+       *
+       *  val runDef = DefDef(runSym, _ => Some('{ println("run") }.asTerm))
+       *  val clsDef = ClassDef(cls, parents, body = List(runDef))
+       *  val newCls = Apply(Select(New(TypeIdent(cls)), cls.primaryConstructor), Nil)
+       *  val modVal = ValDef(mod, Some(newCls))
+       *  val modDef = List(modVal, clsDef)
```

Comment:
Having to construct both the val and class manually is a bit cumbersome, it might be nicer to have a convenience method to do both at once for this usecase.

Reply:
Added `ClassDef.module` to do this. I placed it there because its signature is similar to `ClassDef.apply`.

Not sure if it should return a tuple `(ValDef, ClassDef)` or a `List[Definition]`.